snuba/query/snql/parser.py

```diff
@@ -1537,29 +1543,67 @@ def parse_snql_query(
         with sentry_sdk.start_span(op="processor", description="treeify_conditions"):
             _post_process(query, [_treeify_or_and_conditions], settings)
 
+        # dummy settings and timer dont matter
+        dummy_settings = HTTPQuerySettings()
+        dummy_timer = Timer("snql_pipeline")
+        res = PostProcessAndValidateQuery().execute(
+            QueryPipelineResult(
+                data=(query, dataset, settings, custom_processing),
+                error=None,
+                query_settings=dummy_settings,
+                timer=dummy_timer,
+            )
+        )
+        if res.error:
+            raise res.error
+        assert res.data  # since theres no res.error, data is guaranteed
+        return res.data
+    except InvalidQueryException:
+        raise
+    except Exception:
+        raise PostProcessingError(query)
+
+
+class PostProcessAndValidateQuery(
+    QueryPipelineStage[
+        tuple[
+            LogicalQuery | CompositeQuery[LogicalDataSource],
+            Dataset,
+            QuerySettings | None,
+            CustomProcessors | None,
+        ],
+        LogicalQuery | CompositeQuery[LogicalDataSource],
+    ]
+):
+    def _process_data(
+        self,
+        pipe_input: QueryPipelineData[
+            tuple[
+                LogicalQuery | CompositeQuery[LogicalDataSource],
+                Dataset,
+                QuerySettings | None,
```

Comment:
you don't need QuerySettings here, thery're accessible in the pipeline stage

Reply:
this also relates to the other comment. QueryPipelineResult expects QuerySettings, I have QuerySettings | None, which is why I pass it in and create the dummy one.

Reply:
well then do `settings=query_settings or HTTPQuerySettings()`

Reply:
done